libs/tools/barista/src/transform.ts

```diff
@@ -236,6 +237,44 @@ export const relativeUrlTransformer: BaPageTransformer = async (source) => {
   return transformed;
 };
 
+export const tocGenerator: BaPageTransformer = async (source) => {
+  const transformed = { ...source };
+  let toc: TOC = { headlines: [] };
+  if (source.toc) {
+    // generate TOC and at to source
+    // Find Headlines and corresponding subheadlines (h2 > h3)
+    transformed.content = runWithCheerio(source.content, ($) => {
+      const headlines = $('h2, h3');
+      let currentH2 = -1;
+      headlines.each((_, headline) => {
+        const headlineId = $(headline).attr('id');
+        const headlineText = $(headline).text();
+        if (headline.tagName === 'h2') {
+          currentH2++;
+          toc.headlines.push({
+            id: headlineId!,
+            headline: headlineText!,
+          });
+        } else if (headline.tagName === 'h3') {
+          // Add subheadlines array when needed
+          if (!toc.headlines[currentH2].children) {
+            toc.headlines[currentH2] = {
+              ...toc.headlines[currentH2],
+              children: [],
+            };
```

Comment:
No need for a spread here, this should suffice
```suggestion
            toc.headlines[currentH2].children = [];
```